plumbing/format/commitgraph/v2/chain.go

```diff
@@ -0,0 +1,112 @@
+package v2
+
+import (
+	"bufio"
+	"io"
+	"path"
+
+	"github.com/go-git/go-billy/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+)
+
+type closers []io.Closer
+
+func (c closers) Close() (err error) {
+	for _, closer := range c {
+		// We must use a defer function here otherwise a panic in one or more of the closers would prevent the remaining closers from being closed.
+		defer func(closer io.Closer) {
+			currentErr := closer.Close()
+			if currentErr != nil {
+				err = currentErr
+			}
+		}(closer) // avoid loop variable capture
+	}
+	return
+}
+
+// OpenChainFile reads a commit chain file and returns a slice of the hashes within it
+//
+// Commit-Graph chains are described at https://git-scm.com/docs/commit-graph
+// and are new line separated list of graph file hashes, oldest to newest.
+//
+// This function simply reads the file and returns the hashes as a slice.
+func OpenChainFile(r io.Reader) ([]string, error) {
+	bufRd := bufio.NewReader(r)
```

Comment:
I'm confused, when would anyone pass a nil reader here? What should we return? `nil, nil` or `nil, io.UnexpectedEOF`?

Reply:
The normal process should never cause that, but ideally we should prevent `panic` in case someone does. returning `nil`, `io.UnexpectedEOF` should be fine.